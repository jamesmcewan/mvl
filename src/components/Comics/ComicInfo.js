import Creator from './Creator'

const ComicInfo = ({ isVisible, setIsVisible, title, creators, ...rest }) => (
  <div>
    <div>
      <h2 className="text-yellow-200 text-xl">{title}</h2>
      <ul className="p-0">
        {creators?.items?.map((creator) => (
          <Creator {...creator} key={creator.name} />
        ))}
      </ul>
      <button onClick={() => setIsVisible(!isVisible)}>close</button>
    </div>
  </div>
)

export default ComicInfo
