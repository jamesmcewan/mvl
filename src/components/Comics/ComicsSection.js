const ComicsSection = ({ children }) => <section>{children}</section>;

export default ComicsSection;
