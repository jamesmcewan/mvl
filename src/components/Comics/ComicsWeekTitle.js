const ComicsWeekTitle = ({ children }) => (
  <h1 class="flex justify-center items-center	m-0">{children}</h1>
)

export default ComicsWeekTitle
