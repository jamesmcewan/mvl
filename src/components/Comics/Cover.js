import ProgressiveImage from 'react-progressive-image'
import getCover from '../../functions/getCover'
import CoverImg from './CoverImg'
import placeHolder from '../../images/placeholder.jpg'

const Cover = ({ thumbnail, title, visible, setVisible }) => (
  <button
    class="p-0 m-0 bg-transparent border-none block w-full"
    onClick={() => setVisible(!visible)}
  >
    <ProgressiveImage src={getCover(thumbnail)} placeholder={placeHolder}>
      {(src) => <CoverImg src={src} alt={title} visible={visible} />}
    </ProgressiveImage>
  </button>
)

export default Cover
