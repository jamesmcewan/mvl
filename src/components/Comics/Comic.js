import { useState } from 'react'
import Cover from '../Cover/Cover'
import ComicDetails from '../ComicDetails/ComicDetails'

const Comic = ({ ...rest }) => {
  const [isVisible, setIsVisible] = useState(false)

  return (
    <div data-selector="comic-wrapper">
      <Cover {...rest} {...{ isVisible, setIsVisible }} />
      <ComicDetails {...rest} {...{ isVisible, setIsVisible }} />
    </div>
  )
}

export default Comic
