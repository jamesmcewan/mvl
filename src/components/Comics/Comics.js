import React from 'react';
import { isArray } from 'util';
import styled from 'styled-components';
import Comic from '../Comic/Comic';

const Section = styled.section`
  padding-top: 0;
  margin: 0 auto;
  text-align: center;
  display: flex;
  flex-flow: row wrap;
  align-items: flex-start;
  justify-content: center;
  order: 2;
`;

const Comics = ({ comics, changeSpotlight, isSpotlightVisible }) => (
  <Section>
    {isArray(comics) &&
      comics.map(comic => (
        <Comic
          {...comic}
          changeSpotlight={changeSpotlight}
          key={comic.id}
          isSpotlightVisible={isSpotlightVisible}
        />
      ))}
  </Section>
);

export default Comics;
