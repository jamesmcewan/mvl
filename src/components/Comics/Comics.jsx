import { useState, useEffect } from 'react'
import getComics from '../../helpers/getComics'
import Loading from '../Loading/Loading'
import Comic from './Comic'

const Comics = () => {
  const [isError, setIsError] = useState(false)
  const [comics, setComics] = useState([])
  const [isLoading, setIsLoading] = useState(true)

  useEffect(() => {
    const comicsData = sessionStorage.getItem(`comics`)

    if (comicsData) {
      setComics(JSON.parse(comicsData))
      setIsLoading(false)
      return
    }

    getComics().then(
      (comics) => {
        sessionStorage.setItem(`comics`, JSON.stringify(comics))
        setComics(comics)
        setIsLoading(false)
      },
      (error) => {
        setIsError(true)
        setIsLoading(false)
      }
    )
  }, [])

  return (
    <>
      {isLoading && <Loading />}
      {!isLoading && isError && <div>something went wrong</div>}
      {!isLoading && !isError && (
        <section className="grid grid-cols-1 sm:grid-cols-2 gap-4 md:grid-cols-3">
          {comics?.map((comic) => (
            <Comic {...comic} key={comic.id} />
          ))}
        </section>
      )}
    </>
  )
}

export default Comics
