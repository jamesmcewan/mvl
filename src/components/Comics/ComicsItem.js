import Comic from '../Comic/Comic'

const ComicsItem = (comic) => <Comic {...comic} key={comic.id} />

export default ComicsItem
