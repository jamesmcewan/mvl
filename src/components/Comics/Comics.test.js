import React from 'react';
import { mount } from 'enzyme';
import Comics from './Comics';
import 'jest-styled-components';

describe('Comics', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Comics />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Comics />);
    expect(wrapper).toMatchSnapshot();
  });
});
