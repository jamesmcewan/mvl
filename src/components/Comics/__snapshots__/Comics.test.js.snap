// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comics should render the correct JSX 1`] = `
.c0 {
  padding-top: 0;
  margin: 0 auto;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

<Comics>
  <styled.section>
    <section
      className="c0"
    />
  </styled.section>
</Comics>
`;
